UKHomeOffice/nmsw-ui
# NMSW-304 Display FAL 5 field level errors

# Ticket

NMSW-304

----

## AC

Given the fileupload FAL 5 endpoint has returned error messages
When those errors are displayed
The cell reference and error message text should match the table below

<html><head></head><body>

Cell | Field | Rule | Error Message
-- | -- | -- | --
A5-99+ | travelDocument | required | Enter P, I or O, or you may enter the full word: P - Passport, I - ID card - for a national ID card, or O - Other - for any other document
A5-99+ | travelDocument | wrong value | Enter travel document as P, I, O or as Passport, ID card, Other
C5-99+ | otherTDtype | require value when 'O' or 'Other' was entered in col A | You entered 'Other' as the travel document type. You must enter the type of travel document, for example SID for Seafarer's Identity Document
C5-99+ | travelDocumentIssuingCountry | required | Enter the 3-letter ISO country code for the issuing country; for example, GBR, SWE, NLD
C5-99+ | travelDocumentIssuingCountry | must be 3 letters | Enter the issuing country as a 3-letter ISO country code; for example, GBR, SWE, NLD
D5-99+ | travelDocumentNumber | required | Enter the number of the travel document
D5-99+ | travelDocumentNumber | only numbers | Enter the number of the travel document using only numbers
D5-99+ | travelDocumentNumber | max characters | Enter the number of the travel document in 35 characters or less
E5-99+ | rankRating | required | Enter the rank, rating or job title; for supernumeraries just put SN or supernumerary, not the job title
E5-99+ | rankRating | max characters | Enter the rank, rating or job title in 35 characters or less
E5-99+ | rankRating | invalid characters | Enter the rank, rating or job title using English letters instead of special characters not recognised
F5-99+ | familyName | required | Enter family name or surname as it appears in the travel document
F5-99+ | familyName | max characters | Enter family name or surname in 35 characters or less
F5-99+ | familyName | invalid characters | Enter family name or surname using English letters instead of special characters not recognised
G5-99+ | givenName | required | Enter all forenames or given names as they appear in the travel document - if the crew member has no forename recorded enter UNKNOWN
G5-99+ | givenName | max characters | Enter all forenames or given names in 35 characters or less
G5-99+ | givenName | invalid characters | Enter forenames using English letters instead of special characters not recognised
H5-99+ | gender | required | Enter M for male, F for female, or X for gender neutral if this is in the travel document
H5-99+ | gender | wrong value | Enter M for male, F for female, or X for gender neutral if this is in the travel document
I5-99+ | dateOfBirth | invalid format | Enter the date of birth in the dd/mm/yyyy format, for example, 12/02/2022
J5-99+ | placeOfBirth | max characters | Enter the place of birth in 35 characters or less
J5-99+ | placeOfBirth | invalid characters | Enter place of birth using English letters instead of special characters not recognised
K5-99+ | nationality | must be 3 letters | Enter the nationality as a 3-letter ISO country code; for example, GBR, SWE, NLD
L5-99+ | travelDocumentExpiry | invalid format | Enter the travel document expiry date in the dd/mm/yyyy format, for example, 12/02/2022

</body></html>

----

## To test

- Log in
- Click on `Report a voyage` button
- Upload a general declaration FAL 1
- Go through the process until you see the task list
- Click on "Crew details including supernumeraries (FAL 5)" link
- Upload the 'Crew details including supernumeraries FAL 5 - blank.xlsx' file attached to the ticket
- > You should be taken to the errors found page and be shown errors that fields are required
- NOTE: You will get a 'none is not an allowed value' for K5 - this is a BE bug and will be fixed at a later point'
- Click re-upload file
- Upload the 'Crew details including supernumeraries FAL 5 - maxCharacters.xlsx' file attached to the ticket
- > You should be taken to the errors found page and be shown errors relating to characters
- Click re-upload file
- Upload the 'Crew details including supernumeraries FAL 5 - invalidCharacters.xlsx' file attached to the ticket
- > You should be taken to the errors found page and be shown errors relating to characters

----

## Notes

- You will get a 'none is not an allowed value' for K5 if it left blank - this is a BE bug and will be fixed at a later point'
- There is another BE bug where column B is NOT erroring if user has typed in O or Other into the column A when it should cause an error, this will be fixed at a later stage
